app/Models/Order.ts

```diff
@@ -0,0 +1,23 @@
+import { DateTime } from 'luxon';
+import { BaseModel, column, belongsTo, BelongsTo } from '@ioc:Adonis/Lucid/Orm';
+import Product from 'App/Models/Product';
+
+export default class Order extends BaseModel {
+    @column({ isPrimary: true })
+    public id: number;
+
+    @column()
```

Comment:
This is nice, how does migrations work on this? Through adonis? And it bases itself off of this code?

Reply:
So, Ideally, you can create a model separately using **node ace make:model Order** and it will generate the model for you, however, I like to create models together with their migration schemas using **node ace make:model Order -m**. The -m flag is for migration for that specific model so under the hood when a model called Order is created, a table called orders is also created, then at that point, I define my table columns in the migration and also map those columns in the respective model.

Reply:
Okok, we are using typeorm now and its nice to have declare the model once and automatic get the migrations out of the model. Then the model is the truth, and there will be no drift between model and migration

Reply:
Yes, I have read about it in Nestjs, for future potential projects.